SpatialGDK/Source/SpatialGDKTests/Private/LocalDeploymentManager/LocalDeploymentManagerTest.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "LocalDeploymentManager.h"
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+#include "SpatialGDKDefaultWorkerJsonGenerator.h"
+#include "SpatialGDKEditorSettings.h"
+
+#include "CoreMinimal.h"
+
+#define LOCALDEPLOYMENT_TEST(TestName) \
+	TEST(Services, LocalDeployment, TestName)
+
+namespace
+{
+	static const double MAX_WAIT_TIME_FOR_LOCAL_DEPLOYMENT_OPERATION = 10.0;
```

Comment:
This will fail if the machine has not yet downloaded the latest runtime / pinned version of spatiald. This is only an issue on first pass though.... What we need to do is ensure the machine this test is being ran on has cached the runtime and spatiald and THEN run the tests. 

Not sure how we can do this other than doing an untested local launch and then a tested one or perhaps finding a way of caching the binaries ahead of time

Reply:
Would you say changing a wait time to 20 seconds be better? The relevant function will not wait for the whole 20 seconds, but return as soon as the action is successful.

Reply:
Can't we just add a command in the CI pipeline that does this before the tests are run? We could add a ticket to ensure this is tracked,

Reply:
I'll create a jira ticket for that and mention it in LocalDeploymentManagerTest.cpp file

Reply:
Added